api/app/controllers/spree/api/address_book_controller.rb

```diff
@@ -0,0 +1,43 @@
+module Spree
+  module Api
+    class AddressBookController < Spree::Api::BaseController
+      # Note: the AddressBook is the resource to think about here, not individual addresses
+
+      before_filter :ensure_api_token
+
+      def mine
+         @user_addresses = current_api_user.user_addresses
+      end
+
+      def update
+        address_params = address_book_params
+        default_flag = address_params.delete(:default)
+        address = current_api_user.save_in_address_book(address_params, default_flag)
+        if address.valid?
+          @user_addresses = current_api_user.user_addresses
+          render :mine, status: :ok
+        else
+          invalid_resource!(address)
+        end
+      end
+
+      def remove
+        current_api_user.remove_from_address_book(params[:address_id])
+        @user_addresses = current_api_user.user_addresses
+        render :mine, status: :ok
+      end
+
+      private
+
+      def address_book_params
+        # TODO id is in the permitted_attributes file, but is gone by this point
```

Comment:
I'm looking into this, but I wanted to get the PR started and hope someone could shortcut this. permitted_address_attributes is initialized with an :id but from looking at things in a debugger it's gone by this point. I hope it's not being mutated somewhere.

Also the added :default field is rather specific to this controller. I don't mind it being here much, but I'm open to reconfiguring it elsewhere. Briefly: this controller somewhat flattens the UserAddress join model so that :default is another field of the address json object over the wire